core/src/main/java/org/apache/brooklyn/core/location/AbstractLocation.java

```diff
@@ -459,7 +459,7 @@ public ConfigBag getLocalBag() {
 
         @Override
         public void addToLocalBag(Map<String, ?> vals) {
-            configBag.putAll(vals);
+            configBag = ConfigBag.newInstanceExtending(configBag, vals);
```

Comment:
I don't understand what this change is fixing. Can you explain the reasoning please?